min_versions.py

```diff
@@ -170,6 +179,39 @@ def find_pkg_mismatches(pkg_name, pkg_version, fnames):
     return pkg_mismatches
 
 
+def test_pkg_mismatch_regex():
+    """
+    Validation function for the package mismatch regex
+    """
+    pkgs = {
+        "numpy": "0.0",
+        "scipy": "0.0",
+        "python": "2.7",
+        "python-version": "2.7",
+        "numba": "0.0",
+    }
+
+    lines = [
+        "Programming Language :: Python :: 3.8",
+        "STUMPY supports Python 3.8",
+        "python-version: ['3.8']",
+        'requires-python = ">=3.8"',
+        "numba>=0.55.2",
+    ]
+
+    for line in lines:
+        match_found = False
+        for pkg_name, pkg_version in pkgs.items():
+            matches = match_pkg_version(line, pkg_name)
+
+            if matches:
+                match_found = True
+                break
+
+        if not match_found:
+            raise ValueError(f"Edge case {line} not matched by regex")
```

Comment:
Maybe change this error message to `f'Package mismatch regex fails to cover/match "{line}"'`